docs/acsengine-code-delivery.md

```diff
@@ -0,0 +1,50 @@
+# ACS Engine code delivery guide
+
+[![ACS Engine](https://azurecomcdn.azureedge.net/mediahandler/acomblog/media/Default/blog/a8f28783-3ddc-4081-a57d-6d97147467bf.png)](https://github.com/azure/acs-engine)
+
+ACS Engine is an open source project to generate ARM (Azure Resource Manager) templates DC/OS, Kubernetes, Swarm Mode clusters on Microsoft Azure.
+ACS Engine code lives in Azure, and it is imperative to ensure its quality.
+This document provides guidelines to achieve that.
+
+## Development pipeline
+
+ACS Engine employs CI system that incorporates Jenkins server, configured to interact with ACS Engine GitHub project.
+A recommended way to contribute to ACS Engine is to fork github.com/Azure/acs-engine project.
+and create a separated branch (a feature branch) for the feature you are working on.
+
+The following steps constitute ACS Engine delivery pipeline
+
+ 1. Complete the current iteration of the code change, and check it into the feature branch
+ 2. Invoke unit test. Return to step (1) if failed.
+```sh
+    $ make ci
+```
+ 3. Create a template. Return to step (1) if failed.
+```sh
+    $ acs-engine generate --api-model kubernetes.json
+```
+ 4. Deploy the template in Azure. Return to step (1) if failed.
+```sh
+    $ az group create --name=<RESOURCE_GROUP> --location=<LOCATION>
+    $ az group deployment create \
+      --name <DEPLOYMENT_NAME> \
+      --resource-group <RESOURCE_GROUP> \
+     --template-file azuredeploy.json \
+     --parameters @azuredeploy.parameters.json
+```
+  5. Create a pull request (PR) from github.com/Azure/acs-engine portal.
+  6. The PR triggers a Jenkins job that 
+  + applies the changes to the HEAD of the master branch
+  + generates multiple ARM templates for different deployment scenarios
+  + simultaneously provisions the clusters based on generated templates in Azure
+  This test might take 20-40 minutes.
+  If the test fails, review the logs. If the failure was caused by your code change, return to step (1).
+  Sometimes the test might fail because of intermittent Azure issues, such as resource unavailability of provisioning timeout. In this case manually trigger Jenkins PR job from your GitHub PR page.
+> **Note:** You may want to suppress triggering the PR builds, for example, when you add/change the documentation, or when you are in the middle of your changes but want to make them visible to others.
+> There are few ways to do so:
+> (1) applying ***in progress*** or ***kind/documentation*** label to your PR
+> (2) adding ***[skip ci]*** to the PR title
+> Make sure to mark your PR prior to submitting it.
```

Comment:
OK, skipping entire PRs isn't super useful, unfortunately. We need to be able to skip by commit. I should have been clearer with the acceptance criteria, sorry.